### DISCUSSION TITLE
Set a default value for TextInput with condition on edit form

### DISCUSSION THREAD

**Discussion by amirmmdz** [filamentphp/filament#13362] [Help]
### Package

Panel builder

### Package Version

v3.0.0

### How can we help you?

Hi i have this function in my filament resource:
```
public static function getStandardItems($property_rating_id = null){
        $items = [];
        array_map(function($item)use(&$items,$property_rating_id){
            $items[] = Forms\Components\TextInput::make('standards'.$item['id'])
                        ->numeric()
                        ->label($item['name'])
                        ->default($property_rating_id ? PropertyRatingScore::where('property_rating_id',$property_rating_id)->where('standard_item_id',$item['id'])?->first()?->score : $item['score'])
                        ->readOnly($item['score'] ? true : false);
        },StandardItem::where('status',1)->get()->toArray());
        return $items;
    }
```
And this is how i use this function in my resource form (PropertyRatingResource is my resource name):
```
Forms\Components\Section::make(__("Property Rating"))
                    ->label(__("Property Rating"))
                    ->schema(PropertyRatingResource::getStandardItems($form->model?->id))
                    ->columns([
                        'xl'=>2
                    ])
```
Now on update page TextInputs that generated by getStandardItems do not have default value that i set with getStandardItems function even when i use dd before return $items; in function to see the result of map , TextInputs have the defaultState property for each component that it means this function works fine
now what is problem in my resource that the text input doesn't have default values on my edit form

**Comment by amirmmdz** [comment]
I found the way
formatStateUsing was the way to solve problem and i change default to formatStateUsing and problem solved

**Reply by leandrocfe** [reply]
Yes, `default()` only works on Create Pages. Edit Pages will always fill the data from the model and you can use `formatStateUsing` to change the value
